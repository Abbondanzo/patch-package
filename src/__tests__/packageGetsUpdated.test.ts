import { initTestProject } from "./testProjects"

describe("patch-package", () => {
  // copy left-pad-breakage repo to temp folder
  const project = initTestProject("left-pad-breakage")

  project.install()

  function patchWasApplied() {
    return project.readFileSync("node_modules", "left-pad", "index.js").includes("patch-package")
  }

  // assert patch was applied to 1.1.1
  it("definitely applies patches", () => {
    expect(patchWasApplied()).toBe(true)
  })
  // bump version to 1.1.2
  it("gives a warning when the patch is applied successfully but the version changed", () => {
    expect(project.spawnSync("yarn", ["add", "left-pad@1.1.2"]).stderr.toString()).toMatchSnapshot()

    expect(patchWasApplied()).toBe(true)
  })
  // bump version to 1.1.3
  it("raises an error when the patch is applied", () => {
    const result = project.spawnSync("yarn", ["add", "left-pad@1.1.3"], {
      logStdErrOnError: false,
      throwOnError: false,
    })
    expect(result.stderr.toString()).toMatchSnapshot()
    expect(patchWasApplied()).toBe(false)
  })
})
