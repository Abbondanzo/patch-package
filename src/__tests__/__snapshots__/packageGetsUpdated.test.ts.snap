// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patch-package gives a warning when the patch is applied successfully but the version changed 1`] = `
"
Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    left-pad@1.1.1

  applied to

    left-pad@1.1.2

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package left-pad

  to update the version in the patch file name and make this warning go away.


┌─────────────────────────────────────────────────────────┐
│           [33m patch-package update check failed [39m           │
│           Try running with [36msudo[39m or get access           │
│          to the local update config store via           │
│[36m sudo chown -R $USER:$(id -gn $USER) /Users/dshe/.config [39m│
└─────────────────────────────────────────────────────────┘
"
`;

exports[`patch-package raises an error when the patch is applied 1`] = `
"
**ERROR** Failed to apply patch for package left-pad

  This error was caused because left-pad has changed since you
  made the patch file for it. This introduced conflicts with your patch,
  just like a merge conflict in Git when separate incompatible changes are
  made to the same piece of code.

  Maybe this means your patch file is no longer necessary, in which case
  hooray! Just delete it!

  Otherwise, you need to manually fix the patch file. Or generate a new one

  To generate a new one, just repeat the steps you made to generate the first
  one, but accounting for the changes in left-pad.

  i.e. make changes, run \`patch-package left-pad\`, and commit.

  To manually fix a patch file, Run:

     patch -p1 -i patches/left-pad+1.1.1.patch --verbose --dry-run

  To list rejected hunks. A 'hunk' is a section of patch file that describes
  one contiguous area of changes. They are numbered from 1 and begin with lines
  that look like this:

    @@ -48,5 +49,6 @@ function foo(bar) {

  Remove the conflicting hunks, then manually edit files in

    node_modules/left-pad

  to reflect the changes that the conflicting hunks were supposed to make.

  Then run \`patch-package left-pad\`

  Info:
    Patch was made for version 1.1.1
    Meanwhile node_modules/left-pad is version 1.1.3


┌─────────────────────────────────────────────────────────┐
│           [33m patch-package update check failed [39m           │
│           Try running with [36msudo[39m or get access           │
│          to the local update config store via           │
│[36m sudo chown -R $USER:$(id -gn $USER) /Users/dshe/.config [39m│
└─────────────────────────────────────────────────────────┘
error Command failed with exit code 1.
"
`;
